### PR TITLE
Add network cases of port isolated

### PR DESCRIPTION
**Pull request by chloerh**
- Test vm interface with port isolated settings
- Test network with port isolated settings
- Test attaching interface with port isolated settings
- Test live update interface with port isolated settings

Signed-off-by: Haijiao Zhao <haizhao@redhat.com>

### CODE REVIEW

**Comment by chloerh** [comment]
测试名称 | 运行时间 | 结果
-- | -- | --
multivms.port_isolated.iface.attach_iface.set_yes | 6 分 7 秒 | Fixed
multivms.port_isolated.iface.set_iface.with_br.set_all | 4 分 18 秒 | Passed
multivms.port_isolated.iface.set_iface.with_net.set_all | 4 分 23 秒 | Passed
multivms.port_isolated.iface.set_iface.with_net.set_one | 4 分 8 秒 | Passed
multivms.port_isolated.iface.update_iface.none_to_yes | 3 分 45 秒 | Passed
multivms.port_isolated.iface.update_iface.yes_to_no | 3 分 43 秒 | Passed
multivms.port_isolated.iface.update_iface.yes_to_none | 3 分 43 秒 | Passed
multivms.port_isolated.network.set_no | 4 分 11 秒 | Passed
multivms.port_isolated.network.set_yes | 4 分 12 秒 | Passed

**Comment by chloerh** [comment]
@yalzhang

**Comment by chloerh** [comment]
This pr depends on https://github.com/avocado-framework/avocado-vt/pull/2705

**Comment by chloerh** [comment]
@kylazhang Would you mind taking a look?

File: libvirt/tests/cfg/virtual_network/virtual_network_multivms.cfg [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    expect_ping_vm = 'yes'
+    out_ip = www.redhat.com
+    variants:
+        - port_isolated:
+            feature = 'port_isolated'
+            variants:
+                - iface:
+                    variants:
+                        - set_iface:
+                            case = 'set_iface'
```
Comment (kylazhang) [inline libvirt/tests/cfg/virtual_network/virtual_network_multivms.cfg:17]: How about remove this param and just set the following case = set_iface_network and case = set_iface_bridge?

File: libvirt/tests/src/virtual_network/virtual_network_multivms.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    Test network/interface function on 2 vms:
+
+        - Test settings on 2 vms
+        - Run ping check on 2 vms including pinging each other
+        ...
+
+    """
+    vms = params.get('vms').split()
+    vm_list = [env.get_vm(v_name) for v_name in vms]
+    if len(vm_list) > 2:
```
Comment (kylazhang) [inline libvirt/tests/src/virtual_network/virtual_network_multivms.py:92]: Do we also need to check the < 2 situation? Or to say if !=2 then cancel?

File: libvirt/tests/src/virtual_network/virtual_network_multivms.py [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+                    new_iface.xml = libvirt.modify_vm_iface(
+                        vm_i.name, 'get_xml',
+                        {'port': params.get('new_iface_port')}
+                    )
+                    new_ifaces[vm_i.name] = new_iface
+
+                    # Remove current ifaces on vm
+                    vmxml_i = vm_xml.VMXML.new_from_inactive_dumpxml(vm_i.name)
+                    vmxml_i.xmltreefile.remove_by_xpath('/devices/interface', remove_all=True)
+                    vmxml_i.sync()
```
Comment (kylazhang) [inline libvirt/tests/src/virtual_network/virtual_network_multivms.py:189]: How about use remove_all_device_by_type?

File: libvirt/tests/cfg/virtual_network/virtual_network_multivms.cfg (hunk 1 above)
Comment (chloerh) [reply libvirt/tests/cfg/virtual_network/virtual_network_multivms.cfg:17]: Oh, I understand. I was just hoping to make adding other patterns easier when there should be one. I'd like to keep the way it is if it's ok with you?

File: libvirt/tests/src/virtual_network/virtual_network_multivms.py (hunk 3 above)
Comment (chloerh) [reply libvirt/tests/src/virtual_network/virtual_network_multivms.py:189]: :+1: updated

File: libvirt/tests/src/virtual_network/virtual_network_multivms.py (hunk 2 above)
Comment (chloerh) [reply libvirt/tests/src/virtual_network/virtual_network_multivms.py]: You are right :+1: updated

File: libvirt/tests/cfg/virtual_network/virtual_network_multivms.cfg (hunk 1 above)
Comment (kylazhang) [reply libvirt/tests/cfg/virtual_network/virtual_network_multivms.cfg:17]: That's OK

**Review by kylazhang — APPROVED** [review]

**Pull request merged**